### PR TITLE
Add new unit tests for flat_field

### LINKED ISSUE #3702: Add new unit tests for flat_field
Issue [JP-807](https://jira.stsci.edu/browse/JP-807) was created by Philip Hodge:

A few more unit tests for the flat_field step would be useful.

### PR DESCRIPTION
**Pull request by philhodge**
New unit tests were added for `flat_field`, and some tests were split into multiple functions.

Closes #3702

### CODE REVIEW

File: jwst/flatfield/tests/test_g_average.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import numpy as np
+
+from jwst.flatfield.flat_field import g_average
+
+def evaluate_polynomial(x, coeff):
+    """Evaluate a polynomial."""
+    n = len(coeff)
+    sum = np.zeros_like(x) + coeff[0]
+    for i in range(1, n):
+        sum = sum * x + coeff[i]
```
Comment (hbushouse) [inline jwst/flatfield/tests/test_g_average.py]: Shouldn't the "*" and "+" operators be swapped in this expression, i.e. multiply x by coeff[i] and add the result to sum?

Comment (philhodge) [reply jwst/flatfield/tests/test_g_average.py]: No, but I do have the coefficients in an unusual order.  Here is an example of what I was thinking:

    sum = c[0]
    sum = c[0] * x + c[1]
    sum = (c[0] * x + c[1]) * x + c[2]
    sum = (c[0] * x**2 + c[1] * x + c[2]) * x + c[3]
           = c[0] * x**3 + c[1] * x**2 + c[2] * x + c[3]

I think this is consistent with the way I took the integral by modifying the coefficients and evaluating the polynomial, but I'll do some further checking.

Comment (hbushouse) [reply jwst/flatfield/tests/test_g_average.py]: Oh, I see. OK, I guess that makes sense when it's spelled out in detail like that.

**Comment by nden** [comment]
It looks fine but I wonder why you did not use existing functions for polynomial evaluation and integration, e.g. he ones available in numpy. I think we should try to reuse as much code as possible. If nothing else it reduces the code review time.

**Comment by nden** [comment]
Using modeling is fine, just want to mention that numpy also has polynomial functions now.

`np.polynomial.polynomial.polyval`, `np.polynomial.polynomial.polyint` and others

**Pull request merged**